app/javascript/src/controller_branches.js

```diff
@@ -151,6 +156,7 @@ BranchesController.createBranch = function($node) {
     template_condition: this._branchConditionTemplate,
     event_on_create: function(branch) {
```

Comment:
**Scope for code cleanup/improvement??**

Quite possibly, this earlier method of making things happen on certain events is now becoming redundant or outclassed by the method you've just used for this code change...
`view.$document.on('branchRemoved', function`

To avoid using two different methods of triggering event handling, I'd be voting for one or the other.

This old method clutters the config so, there would be an argument for updating it to fit with a similar approach to what you've done on BranchRemove, while we have the opportunity to change code in this file.